op-e2e/e2eutils/fakebeacon/blobs.go

```diff
@@ -0,0 +1,194 @@
+package fakebeacon
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// FakeBeacon presents a beacon-node in testing, without leading any chain-building.
+// This merely serves a fake beacon API, and holds on to blocks,
+// to complement the actual block-building to happen in testing (e.g. through the fake consensus geth module).
+type FakeBeacon struct {
```

Comment:
I wonder if it's worth it to add tests that basically just confirm that a `FakeBeacon` works similarly to a real beacon-node.